src/utils/fetch.ts

```diff
@@ -0,0 +1,117 @@
+import { RetriableTask, TransientError, withRetries } from 'italia-ts-commons/lib/tasks';
+
+import { Millisecond } from 'italia-ts-commons/lib/units';
+import nodeFetch from 'node-fetch';
+
+// The old following import has been substituted by an explicit declaration
+// { fetchMaxRetries, fetchTimeout } from "../config";
+
+const fetchMaxRetries = 5;
+const fetchTimeout: Millisecond = 1000 as Millisecond;
+
+//
+// Returns a fetch wrapped with timeout and retry logic
+//
+
+function retryingFetch(fetchApi: typeof fetch, timeout: Millisecond, maxRetries: number): typeof fetch {
+  // a fetch that can be aborted and that gets cancelled after fetchTimeoutMs
+  const abortableFetch = AbortableFetch(fetchApi);
+  const timeoutFetch = toFetch(setFetchTimeout(timeout, abortableFetch));
+  // configure retry logic with default exponential backoff
+  // @see https://github.com/pagopa/io-ts-commons/blob/master/src/backoff.ts
+  const exponentialBackoff = calculateExponentialBackoffInterval();
+  const retryLogic = withRetries<Error, Response>(maxRetries, exponentialBackoff);
+  const retryWithTransient429s = retryLogicForTransientResponseError((_: any) => _.status === 429, retryLogic);
+  return retriableFetch(retryWithTransient429s)(timeoutFetch);
+}
+
+//
+// Default fetch configured with a short timeout and an exponential backoff
+// retrying strategy - suitable for calling the backend APIs that are supposed
+// to respond quickly.
+
+export function defaultRetryingFetch(timeout: Millisecond = fetchTimeout, maxRetries: number = fetchMaxRetries) {
+  // Override default react-native fetch with whatwg's that supports aborting
+
+  // NOTE: In fact, react-native supports Aborting Controller. Please check
+  // https://github.com/facebook/react-native/blob/5e36b0c6eb2494cefd11907673aa018831526750/RNTester/js/XHRExampleAbortController.js
+  // Despite being defined in whatwg specs, the abort controller is defined in a different package.
+  // As a result, the import of whatwg could be avoided
+
+  // eslint-disable-next-line functional/immutable-data
+  (global as any).AbortController = require('abort-controller');
+
+  // NOTE: I left the following line commented since loading the custom fetch module ./whatwg-fetch.js
+  // is the approach of the IO APP. The only benefit of the custom file is to bind the "global"
+  // variable, which is required by Typescript. The package node-fetch alreday resolves
+  // such issue, so I switched to it
+
+  // require('./whatwg-fetch');
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any,functional/immutable-data
+  (global as any).fetch = nodeFetch;
+
+  return retryingFetch((global as any).fetch, timeout, maxRetries);
+}
+
+//
+// Fetch with transient error handling. Handle error that occurs once or at unpredictable intervals.
+//
+
+//
+// Fetch with transient error handling. Handle error that occurs once or at unpredictable intervals.
+//
+export function retryLogicForTransientResponseError(
```

Comment:
This function can be moved to io-ts-commons, would you make a PR? Cc @balanza

Reply:
Yes, of course